src/java/org/apache/cassandra/auth/AuthConfig.java

```diff
@@ -69,23 +70,16 @@ public static void applyAuth()
 
         // authorizer
 
-        IAuthorizer authorizer = new AllowAllAuthorizer();
-
-        if (conf.authorizer != null)
-            authorizer = FBUtilities.newAuthorizer(conf.authorizer);
+        IAuthorizer authorizer = authInstantiate(conf.authorizer, AllowAllAuthorizer.class);
 
         if (!authenticator.requireAuthentication() && authorizer.requireAuthorization())
-            throw new ConfigurationException(conf.authenticator + " can't be used with " + conf.authorizer, false);
+            throw new ConfigurationException(conf.authenticator.class_name + " can't be used with " + conf.authorizer, false);
```

Comment:
Would be good to extend this message explaining why

Reply:
Improved the message.